src/js/components/DataTable/buildState.js

```diff
@@ -248,38 +248,43 @@ export const normalizeCellProps = (props, theme) => {
         props?.[propName]?.[context] ||
         // if the propName is used without context, it applies to all contexts
         (tableContextNames.every((n) => !props?.[propName]?.[n]) &&
-          props?.[propName]) ||
+          props?.[propName]);
+      if (value !== undefined) result[context][propName] = value;
+
+      let themeValue =
```

Comment:
I'm not seeing how the theme values make it into the cell when it's not pinned. It seems like this gets dropped now.

It seems like the above value with the theme values still needs to get put in the non-pinned part of the results like it did before. You could still grab the first part without the theme piece and use it at 264 and just tack on the theme pieces (pinned version first)  here directly instead of tacking it on at line 287.